R/moma_expose.R

```diff
@@ -62,22 +62,21 @@ moma_svd <- function(
     }
 
     # Sparsity arguments
-    # "moma_sparsity" includes all penalty types, including fused lasso
-    # group lasso and so on.
-    if (!inherits(u_sparsity, "moma_sparsity") || !inherits(v_sparsity, "moma_sparsity")) {
+    # "_moma_sparsity_type" includes all penalty types
+    if (!inherits(u_sparsity, "_moma_sparsity_type") || !inherits(v_sparsity, "_moma_sparsity_type")) {
```

Comment:
Can we break this into two checks and give a more detailed error (saying whether the problem is `u_sparsity` or `v_sparsity`?

Reply:
We don't plan to expose this function to users. The `moma_svd` function in this file is used as exposing the three methods of C++ class `MoMA` and then testing them.

Reply:
Ok. At some point, I'd still like to put together a diagram to document and see if we can clean up the architecture: there's several entry points and I'm not sure we need them all, but that's for the future.

Reply:
Actually I created a table about these entry points a while ago:
https://github.com/DataSlingers/MoMA/issues/34#issuecomment-512789704

Reply:
Essentially we expose `moma_*lda/cca/pca` to users. All others are for testing purpose.